awx/main/expect/isolated_manager.py

```diff
@@ -241,15 +245,25 @@ def check(self, interval=None):
 
             # discover new events and ingest them
             events_path = self.path_to('artifacts', self.ident, 'job_events')
-            for event in set(os.listdir(events_path)) - handled_events:
-                path = os.path.join(events_path, event)
-                if os.path.exists(path):
-                    event_data = json.load(
-                        open(os.path.join(events_path, event), 'r')
-                    )
-                    event_data.setdefault(self.event_data_key, self.instance.id)
-                    dispatcher.dispatch(event_data)
-                    handled_events.add(event)
+
+            # it's possible that `events_path` doesn't exist *yet*, because runner
+            # hasn't actually written any events yet (if you ran e.g., a sleep 30)
+            # only attempt to consume events if any were rsynced back
+            if os.path.exists(events_path):
+                for event in set(os.listdir(events_path)) - self.handled_events:
+                    path = os.path.join(events_path, event)
+                    if os.path.exists(path):
+                        event_data = json.load(
+                            open(os.path.join(events_path, event), 'r')
+                        )
+                        event_data.setdefault(self.event_data_key, self.instance.id)
+                        dispatcher.dispatch(event_data)
+                        self.handled_events.add(event)
+
+                        # handle artifacts
+                        if event_data.get('event_data', {}).get('artifact_data', {}):
```

Comment:
This block is new, and is necessary to properly persist `set_stat` artifact data.